### ISSUE TITLE
Suggestion.. simplify main UI once unified mode has add labels..

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#245]

**Comment by bbernhard** [comment]
Love that! 

thanks a lot also for the mockup - that looks really cool! ;)

If no bugs occur during the final testing tomorrow, then I'll push the first version of the "unified mode + add labels" to production. That means I could start right after that with the simplification :) 

Just a short question (more questions will for sure follow when I am working on that): should we let the "default (simple) mode" die (at least for non-mobile devices) and make the "browse (expert) mode" the only mode? 

With your new proposed menu structure we would have with this two new top level entries: 

- Unified Mode
- Task Mode

Which submenu entries would the Task Mode have? Should we add every possible combination? i.e:
- Annotate (Default)
- Annotate (Browse)
- Label (Default)
- Label (Browse)
- Validate (Default)
- Validate Browse)

Or does it make sense to get rid of the default mode and make the browse mode the only mode? In that case we could simplify the menu to:

- Annotate
- Label
- Validate

I guess we could make the browse mode more user friendly by adding a "I am feeling lucky" button which displays random tasks. 

Could we get rid off the default mode that way? 

Although I have to admit that the default mode is quite nice for labeling...for labeling I really prefer the default mode over the browse mode. 

Or is there maybe another way how we can simplify the number of menu entries while keeping both the default and the browse mode?

**Comment by dobkeratops** [comment]
> I guess we could make the browse mode more user friendly by adding a "I am feeling lucky" button which displays random tasks.

I wanted to suggest going further than that:- start it with the search box saying “random”, so the first thing you always see is a page of random images ready to click straight into work..  and you just replace the word random with a specific term if you didn’t find an interesting image from the random set.

1 load the site,
2 click “unified search mode..”
3 see random images..  click an interesting image..
4 enter or click a label
5 annotate
6 go to 4,  or get back to the search view..

Right.. unified(search)mode and (automatic) task mode . 
At step 2 some users will go a bit further and enter a specific search.

> Or does it make sense to get rid of the default mode and make the browse mode the only mode? In that case we could simplify the menu to:

Right.. my reasoning is if you want to search, you’re better served with the unified mode; I would make the task mode menu simple, and focus on improving automatic guidance.. that way you can satisfy both casual and expert users. You could swap the order too, show task mode first.

Could validation happen as part of a system guided stream?  I’m sort of hoping validation can eventually be made more interesting by combining with refinement(“what type of car is this? Saloon,suv,not a car..”), or allowing people to do corrections (downvotes?) as they encounter things in unified view

**Comment by dobkeratops** [comment]
Ok I just tried the latest version .. I see the add label box in unified mode.. great to see, the goal is in sight..
Using it on the iPad, I found it didn’t seem to always work , although it did once. (I will just try on the desktop again). There’s also a residual problem where if you delete anything it sometimes won’t let you annotate something else after.

Best to keep the whole menu as you have it until this is rock solid IMO.. once this is solid then it will allow the simplification. It might need the seperator/add hot key aswell.. (it will certainly be pleasant to add a bunch of labels with hands on the keyboard then flick between them with hands on the mouse etc to draw)

**Comment by bbernhard** [comment]
> Using it on the iPad, I found it didn’t seem to always work , although it did once. (I will just try on the desktop again). 

that's strange :/ I haven't tried it on a tablet yet, so maybe there is something broken in the tablet version (will check that next). Would be great if you could give it a try on your desktop machine to check whether it's really just the tablet version that's buggy :) 

> There’s also a residual problem where if you delete anything it sometimes won’t let you annotate something else after.

I'll have a look tomorrow to see whether I can reproduce that in my dev environment. If you recognize a pattern that helps tracking down the bug, please let me know :)


> Best to keep the whole menu as you have it until this is rock solid IMO.. once this is solid then it will allow the simplification. It might need the seperator/add hot key aswell.. (it will certainly be pleasant to add a bunch of labels with hands on the keyboard then flick between them with hands on the mouse etc to draw)

totally agreed. I think we might also need a bunch of hotkeys that make the navigation between the different UI elements a bit easier. (switching between labeling/annotation feels bit sluggish at the moment)

**Comment by dobkeratops** [comment]
Right.. just tried it on the desktop .. works fine! This is a HUGE step forward! This is the best it’s been, trying to think what might be different on an iPad browser.. there are little differences in how it responds to focus (eg it scrolls the list differently), but I would worry so much. The fact this works on a desktop machine or laptop is great. (after breaking my laptop ive been using my desktop a lot more)



 we still need the dedicated add labels screen for label suggestions.

> I think we might also need a bunch of hotkeys that make the navigation between the different UI elements a bit easier. (switching between labeling/annotation feels bit sluggish at the moment)

Even as it is it’s great, but yes if you had similar “add label” hotkeys and seperator detection to the add label screen , and label toggle hotkeys (to work “dpaint style”.. like the palette switch hotkeys) that would be perfect

**Comment by bbernhard** [comment]
> Right.. just tried it on the desktop .. works fine! This is a HUGE step forward! This is the best it’s been, trying to think what might be different on an iPad browser.. there are little differences in how it responds to focus (eg it scrolls the list differently),
...but I would worry so much. The fact this works on a desktop machine or laptop is great. (after breaking my laptop ive been using my desktop a lot more)

Thanks, that's great to hear!

I've just checked with my Pixel C tablet and it seems that there big differences between the various browsers. With firefox (my default browser) the site is almost unusable on the tablet. Drawing a rectangle is really sluggish and slow. There also seems to be a rendering bug when adding a new label in the unified mode (sometimes the newly added label won't show up although it is there; a touch on the labels list usually triggers a re-rendering of that part which fixes the problem). 

With the chrome browser however, everything works fine.

Really strange...

**Comment by dobkeratops** [comment]
Bearing that in mind, perhaps in any UI simplification you do want to keep the various modes available as woerkarounds.. I don’t know what’s easier to maintain, more simpler screens or one sdvanced screen. I could still imagine thededicated “add labels” screen being useful with its large area devoted to.popular label shortcuts (imagine that furthe guided by scene label correlation.. pick a scene or category first, then get label suggestions for that..)

And there is of course phones to consider.. I haven’t tried using it on a smartphone, which is what most people on this planet probably use most of the time now. (I imagine the original task idea is probably a solution to fitting enough work on a smaller screen.. no room for a label palette plus workspace? .. so you probably do want to keep it in parallel..)

Anyway.. lots of options to explore.
